Jenkinsfile

```diff
@@ -0,0 +1,334 @@
+sshkey_dev_preview_int = "test-key"
+sshkey_dev_preview_stg = "test-key"
+sshkey_dev_preview_prod = "test-key"
+
+last_action_day = 0
+year=getCurrentYear()
+
+// TODO: as these fields mature and stop changing, change into be build parameters so that a simple script can trigger a build
+def dates = input(
+        message: 'Enter desired dates for cluster deployments (MM/DD/YYYYY)?\nDates are inclusive. Use 00/00/0000 if this step is not relevant to this run.',
+        parameters: [
+                [$class: 'hudson.model.StringParameterDefinition', defaultValue: '', description: 'Name of this spring', name: 'sprint_name'],
+                [$class: 'hudson.model.StringParameterDefinition', defaultValue: "00/00/${year}", description: 'Date for the teardown and recreate dev-preview-int (first Friday of the sprint)', name: 'int_recreate'],
+                [$class: 'hudson.model.StringParameterDefinition', defaultValue: "00/00/${year}", description: 'Date of first upgrade for dev-preview-int (usually second Monday of the sprint)', name: 'int_upgrades_start'],
+                [$class: 'hudson.model.StringParameterDefinition', defaultValue: "00/00/${year}", description: 'Date of last upgrade for dev-preview-int (usually third Monday of the sprint)', name: 'int_upgrades_stop'],
+                [$class: 'hudson.model.StringParameterDefinition', defaultValue: "00/00/${year}", description: 'Date of first upgrade for dev-preview-stg (usually third Tuesday of the sprint)', name: 'stg_upgrades_start'],
+                [$class: 'hudson.model.StringParameterDefinition', defaultValue: "00/00/${year}", description: 'Date of last upgrade for dev-preview-stg (usually third Friday of the sprint)', name: 'stg_upgrades_stop'],
+                [$class: 'hudson.model.StringParameterDefinition', defaultValue: "00/00/${year}", description: 'Date of upgrade for dev-preview-prod (usually first Monday of \'next\' sprint); This process will not be performed without user input.', name: 'prod_upgrade'],
+        ]
+)
+
+echo "Launching sprint ${dates.sprint_name}"
+echo "Dates: ${dates}"
+
+// Make sure dates are valid
+splitDate(dates.int_recreate)
+splitDate(dates.int_upgrades_start)
+splitDate(dates.int_upgrades_stop)
+splitDate(dates.stg_upgrades_start)
+splitDate(dates.stg_upgrades_stop)
+splitDate(dates.prod_upgrade)
+
+
+while ( true ) {
+    def todayFields = getCurrentDateFields()
+
+    phase = "dev-preview-int/create"
+    if ( isToday( phase, todayFields, dates.int_recreate ) ) {
+
+        def skip_build = false
+        while ( true ) {
+
+            try {
+                if ( !skip_build ) {
+                    stage( "${phase} openshift/ose build" ) {
+                        runOSEBuild()
```

Comment:
@stevekuznetsov Yep. That is actually exactly what @tdawson has requested. Moving it to async at 6am with the job build/scheduled/ose-3.5 . The only drawback is that the sprint-control script will send out a "success" note if the build fails -- because the old level is already installed and upgrade will do nothing. Lots of room for improvement.